src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -123,19 +122,8 @@ public void initTagPositions()
             return;
         }
 
-        final IBlueprintDataProvider te = getTileEntity();
-        if (te != null)
-        {
-            initTags = true;
-            sitPositions = new ArrayList<>();
-            for (final Map.Entry<BlockPos, List<String>> entry : te.getWorldTagPosMap().entrySet())
-            {
-                if (entry.getValue().contains(TAG_SITTING))
-                {
-                    sitPositions.add(entry.getKey());
-                }
-            }
-        }
+        sitPositions = getLocationsFromTag(TAG_SITTING);
+        initTags = !sitPositions.isEmpty();
```

Comment:
Actually, initTags is always true if the TE was not null

Reply:
Yes, that's the semantic change I mentioned above.  `te != null` is not visible here any more, so this is the next best thing since `getLocationsFromTag` will return an empty list if the te is null.

Since the te should never be null here anyway unless something is broken, setting unconditionally to true may be reasonable behaviour too.  Unless we're expecting the building to exist before the te does?

Reply:
The TE is null if the hut is not loaded.

Reply:
The differences in actual behaviour between these options:
* With previous implementation, it will keep re-checking as long as te==null, but afterwards stop even if the list is empty.
* With current implementation, it will keep re-checking the tags each time this is called until at least one tag exists.  This allows the player to add the tag themselves after building the structure, if it was missing for some reason.  But possibly has a slight performance hit to keep re-checking if this is called too often.
* With "just set to true", it will give up immediately if te == null or tags is empty on first call after build/repair/upgrade.

Reply:
technically, you coudl return null fromt he getLocations method if the TE is null.

Reply:
True, and I could change it to that if the distinction is meaningful.  Since it's a convenience method, I wanted to be convenient and not return null though.